### PR TITLE
Nearby events list mode

### PR DESCRIPTION
**Pull request by DavidKalajdzic**
- The EventsActivity has now the map and list view modes, all data is fetched from Firebase. Only those that have positions will be visible in Map mode, whereas all events are listed in the List mode.
- The UI is responsive
- From the list we can open profile page of the organiser by clicking on his profile image
- Reorganised all the files into coherent subfolders in `event` folder

### CODE REVIEW

File: app/src/main/java/com/sdpteam/connectout/event/creator/LocationPicker.java [hunk 1]
```diff
@@ -12,26 +12,28 @@
...
 import com.google.android.gms.maps.model.MarkerOptions;
 import com.sdpteam.connectout.event.Event;
-import com.sdpteam.connectout.event.EventsViewModel;
+import com.sdpteam.connectout.event.nearbyEvents.EventsViewModel;
+import com.sdpteam.connectout.event.nearbyEvents.map.EventsMapViewFragment;
 
 import androidx.annotation.NonNull;
 
-public class PositionSelectorFragment extends MapViewFragment implements OnMapReadyCallback {
+// TODO : it is very strange to make a fragment extend another one. especially when they have different purposes.
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/event/creator/LocationPicker.java:20]: Let's open an issue to refactor this independently of this PR

**Review by MikaelKalajdzic — APPROVED** [review]
Thank you for the renamings and package reordering

File: app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+/**
+ * It is responsible for creating and managing the views for a list of events.
+ * The class holds the context, resource, and a list of events, which are passed in through the constructor.
+ */
+public class EventsAdapter extends ArrayAdapter<Event> {
+
+    private final Context context;
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsAdapter.java:27]: Don't need to keep the context here, you can access it from the superclass with getContext() ;)

File: app/src/main/java/com/sdpteam/connectout/event/creator/LocationPicker.java (hunk 1 above)
Comment (DavidKalajdzic) [reply app/src/main/java/com/sdpteam/connectout/event/creator/LocationPicker.java:20]: https://github.com/ConnectOut-sdp/sdp2023/issues/58

**Pull request merged**